extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/config/ConfigUtils.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.arc.runtime.config;
+
+import java.util.Collection;
+
+import org.eclipse.microprofile.config.Config;
+
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.StringUtil;
+
+public final class ConfigUtils {
```

Comment:
This class shouldn't be probably public...

Reply:
It is being called from the generated bytecode. I could of couse move all of it's functionality into generated code. Should I do that?

Reply:
We could introduce an abstract super type (`AbstractConfigProperties`) and move the functionality into a protected method?

Reply:
Nice, I like that solution a lot! I'll implement it once I'm done trying to integrate validation into this work

Reply:
I am getting illegal access error all over the place even when I place the generated classes in the same package. 
The only way I don't get the illegal access errors is if the class and method is public - which is exactly what we don't want.

So I'm going to just push the functionality of this method into the generated bytecode.

Reply:
Hm, that's odd - `protected` should not care about the package at all?

Reply:
Agreed... But I got fed up trying  to fight it (only public worked) so I will just generate it and all problems will go away (plus it will be totally hidden)

Reply:
I ended up generating the a utility class not an abstract superclass because I was getting a warning from `BeanArchives.index`.

So now everything should be nicely hidden from the user and the functionality is back to what it was